go/marshal/decode.go

```diff
@@ -38,16 +38,16 @@ import (
 // into corresponding Go array elements. If the Go map was nil a new map is
 // created if any value is set.
 //
-// To unmarshal Noms sets, it depends on the presence of a `noms:",set"` tag:
-//  - Without (default), Unmarshal decodes into corresponding Go list elements.
-//  - With, Unmarshal decodes into Go map keys corresponding to the set values,
-//    with values struct{}{}. Map values must have struct{} type.
+// To unmarshal a Noms set into a Go map, the field must be tagged with `noms:",set"`,
+// and it must have a type of map[<value-type>]struct{}. Unmarshal decodes into
+// Go map keys corresponding to the set values and assigns each key a value of struct{}{}.
```

Comment:
As discussed, the code already handled unmarshalling into slices and arrays regardless of annotation. It would be nice to produce an error if a slice/array is annotated as a set but the noms types is a list, but I don't think I'm going to do that in this patch.